src/qt/walletmodel.cpp

```diff
@@ -514,6 +514,8 @@ bool WalletModel::bumpFee(uint256 hash, uint256& new_hash)
     questionString.append("</td><td>");
     questionString.append(BitcoinUnits::formatHtmlWithUnit(getOptionsModel()->getDisplayUnit(), new_fee));
     questionString.append("</td></tr></table>");
+    questionString.append("<br><br>");
+    questionString.append(tr("WARNING: New inputs and outputs can be added in replacement transaction if necessary which affects privacy."));
```

Comment:
grammar nit, "which" needs a preceding comma
```suggestion
    questionString.append(tr("WARNING: New inputs and outputs can be added in replacement transaction if necessary, which affects privacy."));
```

Reply:
An article ("a", etc.) is also missing before the word "replacement"

Reply:
"can be added...if necessary" is vague, would suggest being more precise: added by what, why would it be necessary.

Reply:
> An article ("a", "the", etc.) is also missing before the word "replacement"

Will add "the"

> "can be added...if necessary" is vague, would suggest being more precise: added by what, why would it be necessary.

I rephrased _'adding inputs when necessary. It may add a new change output if one does not already exist.'_ from https://bitcoincore.org/en/doc/0.20.0/rpc/wallet/bumpfee/ Not sure what else can I add which even normal users can understand

Reply:
Thanks for updating! (very kind but no need to credit me for a drive-by comment) can you squash to one commit?

Reply:
Oh I see, sorry, this is based somewhat on the bumpfee help.
```
The command will pay the additional fee by reducing change outputs or adding inputs when necessary.
It may add a new change output if one does not already exist.
All inputs in the original transaction will be included in the replacement transaction.
```

Reply:
I'm not sure this should have a screaming WARNING.

Maybe:

```
This will pay the additional fee by reducing change outputs
or adding inputs when necessary. It may add a new change output
if one does not already exist. These changes may affect privacy.
```

Reply:
> can you squash to one commit?

Sure will do once we finalize the warning message. Does the below mentioned thing look okay?

**WARNING:** This will pay the additional fee by reducing change outputs
or adding inputs when necessary. It may add a new change output
if one does not already exist. These may potentially leak privacy.

Reply:
But the message will be incorrect because the issue is more than just user not aware of how RBF works in bitcoin core wallet. It is also the coin selections algorithm when using a custom change address in the initial transaction. Original Issue: https://github.com/bitcoin/bitcoin/issues/20795

Reply:
No extra inputs or outputs were added in this: https://tbtc.bitaps.com/f35a69b066d1ff79cb13bc6dba2eb345f239b2bfa268e3fb0ee06b28748bea99 when I didn't use "custom change address" for initial transaction

Reply:
If you `setlabel` the change address before bumping the fee it causes Core not to recognize the change address as a change address and to therefore add a new input. I think it makes sense to label all your change UTXOs if you plan to use coin control for privacy. Without labelling your change how do you hope to prevent undesired UTXO linkage?

I wrote about this in bitcoin/bitcoin#20935.

Reply:
I could fix the issue but I doubt it would see enough review; my fee rate work has just been sitting there for the past couple months.

As for the message, could s/will/may/, e.g.:

"This may pay the additional fee by reducing change outputs or adding inputs, when necessary. It may add a new change output if one does not already exist. These may potentially leak privacy."

Reply:
@dooglus Thanks for sharing. Will do some test and respond in https://github.com/bitcoin/bitcoin/issues/20935

@jonatack Made changes in https://github.com/bitcoin-core/gui/commit/570ebc6238639d40c7d54bbf39857de9416f2ba2